[REVIEW] Class label manipulation

tfeher opened the pull request:
Some classifiers work with +/-1 labels, this MR adds functions to query the unique labels and relabel the input data to +/-1.

I have placed it in the "array" subfolder, which had already some label manipulation functions. I think calling the folder "array" is not descriptive. Would it be fine to change the folder name to "labels"?

GPUtester commented:
Can one of the admins verify this patch?

cjnolet commented:
I agree completely about the use of the “array” instead of a name more descriptive of its actual use. I created that folder to mirror the “matrix” folder but I am in favor of changing it to “label”. You are welcome to do the rename in this PR.

cjnolet commented:
add to whitelist

teju85 commented on cpp/src_prims/label/classlabels.h:
```diff
@@ -0,0 +1,116 @@
+  size_t bytes = 0;
+  size_t bytes2 = 0;
+  int *d_num_selected_out;
+  d_num_selected_out = (int *)allocator->allocate(1 * sizeof(int), stream);
```
Better to use `device_buffer`, no?

tfeher commented:
@cjnolet thanks for the comments, I have moved make_monotonic into classlabels.h, and simplified it by using one of the new functions. Please have a look.
